datalad/distribution/utils.py

```diff
@@ -214,7 +215,43 @@ def _install_subds_from_flexible_source(ds, sm_path, sm_url, reckless):
     # do fancy update
     if sm_path in ds.get_subdatasets(absolute=False, recursive=False):
         lgr.debug("Update cloned subdataset {0} in parent".format(subds))
+        # TODO: move all of that into update_submodule ??
+        # TODO: direct mode ramifications?
```

Comment:
started to move into update_submodule but that one is defined in GitRepo, and so far nothing decorated at AnnexRepo level.  @bpoldrack -- should it work currently in direct mode somehow?